packages/techdocs-common/src/stages/publish/azureBlobStorage.ts

```diff
@@ -220,16 +217,28 @@ export class AzureBlobStoragePublish implements PublisherBase {
   ): Promise<TechDocsMetadata> {
     const entityRootDir = `${entityName.namespace}/${entityName.kind}/${entityName.name}`;
     try {
-      return await new Promise<TechDocsMetadata>(resolve => {
-        const download = this.download(
-          this.containerName,
-          `${entityRootDir}/techdocs_metadata.json`,
-        );
-        resolve(JSON5.parse(download.toString()));
+      return await new Promise<TechDocsMetadata>(async (resolve, reject) => {
+        try {
+          const techdocsMetadataJson = await this.download(
+            this.containerName,
+            `${entityRootDir}/techdocs_metadata.json`,
+          );
+          if (!techdocsMetadataJson) {
+            throw new Error(
+              `Unable to parse the techdocs metadata file ${entityRootDir}/techdocs_metadata.json.`,
+            );
+          }
+          const techdocsMetadata = JSON5.parse(
+            techdocsMetadataJson.toString('utf-8'),
+          );
+          resolve(techdocsMetadata);
+        } catch (err) {
+          this.logger.error(err.message);
```

Comment:
Avoid log + throw, let the caller handle the error by logging if they want to

Reply:
So, I better remove `this.logger.error(err.message)` and just leave the `reject(new Error (err.message))`?

Reply:
Just `reject(err)` should be fine, it's already an error - unless you explicitly want to translate the exception.